tokio-executor/src/task/list.rs

```diff
@@ -0,0 +1,88 @@
+use crate::task::{Header, Task};
+
+use std::fmt;
+use std::ptr;
+
+pub(crate) struct OwnedList<T: 'static> {
+    head: *const Header<T>,
```

Comment:
TIOLI: Consider `Option<NonNull<Header<T>>>`? Should have the same repr but would allow some of the `<*const T>::as_ref()` calls to be removed...